src/pages/ContributorFields.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { FC, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { TextField, Typography } from '@material-ui/core';
+import { emptyContributor, Resource } from '../types/resource.types';
+import { ErrorMessage, Field, FieldArray, FieldProps, FormikProps, FormikValues } from 'formik';
+import Button from '@material-ui/core/Button';
+import { createContributor, putContributorFeature } from '../api/resourceApi';
+import Paper from '@material-ui/core/Paper';
+import styled from 'styled-components';
+import DeleteIcon from '@material-ui/icons/Delete';
+import AddIcon from '@material-ui/icons/Add';
+
+const StyledPaper = styled(Paper)`
+  width: 100%;
+  padding: 3rem;
+`;
+
+const StyledField = styled(Field)`
+  display: inline-block;
+  margin-left: 2rem;
+  margin-right: 2rem;
+  padding: 1rem;
+`;
+const StyledDiv = styled.div`
+  display: inline-block;
+  margin-top: 1rem;
+  margin-bottom: 0.3rem;
+`;
+
+const StyledTextField = styled(TextField)`
+  width: 22rem;
+  margin-left: 0.5rem;
+  margin-right: 0.5rem;
+`;
+
+const StyledButton = styled(Button)`
+  position: relative;
+  vertical-align: bottom;
+`;
+
+interface ContributorFieldsProps {
+  resource: Resource;
+  formikProps: FormikProps<FormikValues>;
+  setAllChangesSaved: any;
+}
+
+const ContributorFields: FC<ContributorFieldsProps> = ({ resource, formikProps, setAllChangesSaved }) => {
+  const { t } = useTranslation();
+  const [reloadState, setReloadState] = useState(false);
+
+  const addContributor = () => {
+    createContributor(resource.identifier).then((contributorResponse) => {
+      resource.contributors
+        ? resource.contributors.push(contributorResponse.data)
+        : (resource.contributors = [contributorResponse.data]);
+      // Hacky way to force ContributorFields to update:
+      setReloadState(!reloadState);
+    });
+  };
+
+  const saveField = async (
+    event: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>,
+    resetForm: any,
+    currentValues: FormikValues
+  ) => {
+    setAllChangesSaved(false);
+    if (resource) {
```

Comment:
resource har vi vel uansett her - men jeg tror ikke den trengs. Bruk heller formik-values for å hente ut verdier - vi har ingen garantier for at resource-objektet blir oppdatert

Reply:
Fikset